watcher/src/watcher.rs

```diff
@@ -110,6 +113,15 @@ impl Watcher {
             .unwrap_or(0))
     }
 
+    /// Collect number of blocks synced per peer and ledger height (if
+    /// available)
+    pub fn collect_metrics(&self, ledger_height: Option<u64>) {
+        let last_synced = self.watcher_db.last_synced_blocks().unwrap_or_default();
+        self.metrics.collect_peer_blocks_synced(last_synced);
```

Comment:
Optional/Equivalent: Use conditional if like you do for `ledger_height`
```suggestion
        if let Some(last_synced) = self.watcher_db.last_synced_blocks() {
            self.metrics.collect_peer_blocks_synced(last_synced);
        }
```

Reply:
This is one actually wouldn't work with Some because the return type is a Result, not Some, and the unwrap_or_default is meant to cover the result type.

Reply:
Then you could write it as `if let Ok(last_synced) = ...`